Errors: 
11:38:57 [ERROR]   OnnxOperationsTestCase.testConcatReduce8:141->assertEval:654->assertEval:674->evaluate:622->renameToStandardType:722 » IllegalArgument cell address (2,0,0) is not within the bounds of tensor(d0[2],d1[3],d2[4])
11:38:57 [ERROR]   OnnxOperationsTestCase.testExpand8:580->assertEval:650->assertEval:674->evaluate:622->renameToStandardType:722 » IllegalArgument cell address (2,0,0) is not within the bounds of tensor(d0[2],d1[3],d2[4])
11:38:57 [ERROR]   OnnxOperationsTestCase.testUnsqueeze1:372->assertEval:638->assertEval:674->evaluate:622->renameToStandardType:722 » IllegalArgument cell address (1,0,0,0) is not within the bounds of tensor(d0[1],d1[1],d2[2],d3[3])

--- a/vespajlib/src/main/java/com/yahoo/tensor/functions/Rename.java
+++ b/vespajlib/src/main/java/com/yahoo/tensor/functions/Rename.java
@@ -89,7 +89,7 @@ public class Rename<NAMETYPE extends Name> extends PrimitiveTensorFunction<NAMET
         for (int i = 0; i < tensor.type().dimensions().size(); i++) {
             String dimensionName = tensor.type().dimensions().get(i).name();
             String newDimensionName = fromToMap.getOrDefault(dimensionName, dimensionName);
-            toIndexes[i] = renamedType.indexOfDimension(newDimensionName).get();
+            toIndexes[renamedType.indexOfDimension(newDimensionName).get()] = i;
         }
 
         // avoid building a new tensor if dimensions can simply be renamed
@@ -100,7 +100,7 @@ public class Rename<NAMETYPE extends Name> extends PrimitiveTensorFunction<NAMET
         Tensor.Builder builder = Tensor.Builder.of(renamedType);
         for (Iterator<Tensor.Cell> i = tensor.cellIterator(); i.hasNext(); ) {
             Map.Entry<TensorAddress, Double> cell = i.next();
-            TensorAddress renamedAddress = rename(cell.getKey(), toIndexes);
+            TensorAddress renamedAddress = cell.getKey().partialCopy(toIndexes);
             builder.cell(renamedAddress, cell.getValue());
         }
         return builder.build();
@@ -116,10 +116,6 @@ public class Rename<NAMETYPE extends Name> extends PrimitiveTensorFunction<NAMET
             }
         }
         return true;
-    }
-
-    private static TensorAddress rename(TensorAddress address, int[] toIndexes) {
-        return address.partialCopy(toIndexes);
     }
 
     private String toVectorString(List<String> elements) {

